src/rebar3_hex_app.erl

```diff
@@ -0,0 +1,205 @@
+%%% ===================================================================
+%%% App utils
+%%% ===================================================================
+-module(rebar3_hex_app).
+
+-export([vcs_vsn/2, find/2, get_deps/1, validate/1]).
+
+-define(VALIDATIONS, [
+    has_semver,
+    has_contributors,
+    has_maintainers,
+    has_description,
+    has_licenses,
+    has_unstable_deps
+]).
+
+-type app_data() :: #{
+    name := string(),
+    version := string(),
+    details := list(),
+    deps := list()
+}.
+
+-type validation_errors() :: #{
+    warnings => list(),
+    errors => list()
+}.
+
+-spec find([rebar_app_info:t()], string()) -> {ok, rebar_app_info:t()} | {error, app_not_found}.
+find(Apps, Name) ->
+    Pred = fun(App) ->
+        rebar_app_info:name(App) =:= rebar_utils:to_binary(Name)
+    end,
+    case lists:filter(Pred, Apps) of
+        [App] ->
+            {ok, App};
+        _ ->
+            {error, app_not_found}
+    end.
+
+-spec get_deps(list()) -> {ok, list()} | {error, {non_hex_deps, list()}}.
+get_deps(Deps) ->
```

Comment:
This was moved here as it is can be said it's a pure function. One of the goals in this refactor was to start moving out pure functions into appropriate modules for better testing (i.e., not having to rely on so called integration tests so much, which currently only really tell us if something blew up or not).